build_projects/shared-build-targets-utils/VersionRepoUpdater.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class VersionRepoUpdater
+    {
+        private static Regex s_nugetFileRegex = new Regex("^(.*?)\\.(([0-9]+\\.)?[0-9]+\\.[0-9]+(-([A-z0-9-]+))?)\\.nupkg$");
+
+        private HttpClient _client = new HttpClient();
+        private string _gitHubUser;
+        private string _gitHubEmail;
+        private string _versionsRepoOwner;
+        private string _versionsRepo;
+
+        public VersionRepoUpdater(
+            string gitHubUser = null, 
+            string gitHubEmail = null, 
+            string versionRepoOwner = null, 
+            string versionsRepo = null, 
+            string gitHubAuthToken = null)
+        {
+            _gitHubUser = gitHubUser ?? "dotnet-bot";
+            _gitHubEmail = gitHubEmail ?? "dotnet-bot@microsoft.com";
+            _versionsRepoOwner = versionRepoOwner ?? "dotnet";
+            _versionsRepo = versionsRepo ?? "versions";
+
+            gitHubAuthToken = gitHubAuthToken ?? Environment.GetEnvironmentVariable("GITHUB_PASSWORD");
+
+            if (string.IsNullOrEmpty(gitHubAuthToken))
+            {
+                throw new ArgumentException("A GitHub auth token is required and wasn't provided. Set 'GITHUB_PASSWORD' environment variable.", nameof(gitHubAuthToken));
+            }
+
+            _client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+            _client.DefaultRequestHeaders.Add("Authorization", $"token {gitHubAuthToken}");
+            _client.DefaultRequestHeaders.Add("User-Agent", _gitHubUser);
+        }
+
+        public async Task UpdatePublishedVersions(string nupkgFilePath, string versionsRepoPath)
+        {
+            List<Tuple<string, string>> publishedPackages = GetPackageInfo(nupkgFilePath);
+
+            string packageInfoFileContent = string.Join(
+                Environment.NewLine,
+                publishedPackages
+                    .OrderBy(t => t.Item1)
+                    .Select(t => $"{t.Item1} {t.Item2}"));
+
+            string firstVersionWithPrerelease = publishedPackages
+                .FirstOrDefault(t => t.Item2.Contains('-'))
+                ?.Item2;
+
+            string prereleaseVersion = null;
+            if (!string.IsNullOrEmpty(firstVersionWithPrerelease))
+            {
+                prereleaseVersion = firstVersionWithPrerelease.Substring(firstVersionWithPrerelease.IndexOf('-') + 1);
+            }
```

Comment:
I haven't decided on that yet.... If we do it would be the full version, not just the pre-release version, like we do for CoreFX.

I think moving the towards the model that you update each individual dependency with its "full new version" is better than the model we started with of updating just the pre-release and leaving the beginning unchanged.

Another issue right now is that CLI publishes two sets of NuGet packages at two different times. 

We can definitely add a Latest.txt file in the future when needed.